### DISCUSSION TITLE
How to request a new package?

### DISCUSSION THREAD

**Discussion by rth238** [Homebrew/discussions#533] [Getting started] [answered]
Sorry if this is a dumb question, but I've not seen how to proceed yet...
I am exploring moving from MacPorts to Homebrew.  A package that I depend on with MacPorts, pgplot, seems to have been supported in the past on Homebrew, but also does not seem to be currently available. (MacPorts version of pgplot is currently broken on BigSur/Apple Silicon.)
Is there a mechanism for asking for the including of a new package for Homebrew?

**Comment by SMillerDev** [comment]
https://docs.brew.sh/Adding-Software-to-Homebrew

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/Homebrew/discussions/533#discussioncomment-279618

@SMillerDev, that doesn't explain how to *request* a new package if you're not technically competent enough to create one.

**Reply by SMillerDev** [reply]
That's the only way. There's plenty of people willing to help once you get started, and who knows... you might surprise yourself

**Comment by carlocab** [comment, marked as answer]
`pgplot` was removed due to license issues. We do have a formula that substitutes most (though not all) of its functionality: `giza`.

You can also host a `pgplot` formula in your own tap if you wish. This will allow you to install it just like any other Homebrew formula, but you'll need to maintain the formula yourself. Here's a copy of the formula that I think will work, but I haven't tested it so I'm not making any promises about it: [pgplot.rb.txt](https://github.com/Homebrew/discussions/files/5810360/pgplot.rb.txt) (`.txt` extension because GitHub doesn't support `.rb` extensions when uploading text files. You can just get rid of the `.txt` part.)



For more on this, see:
https://docs.brew.sh/Taps
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by rth238** [reply]
Thanks!  I'll give this a try.

**Reply by Nokii9** [reply]
A
